src/tir/schedule/primitive/fuse_split.cc

```diff
@@ -0,0 +1,483 @@
+  }
+  PrimExpr substitute_value = 0;
+  for (size_t i = 0; i < inferred_factors.size(); i++) {
+    substitute_value *= inferred_factors[i];
+    substitute_value += new_loop_vars[i];
+  }
+  Map<Block, Block> opaque_block_reuse;
+  auto substitute_function = [&](const Var& v) -> Optional<PrimExpr> {
+    if (v.same_as(loop->loop_var)) {
+      return substitute_value;
+    } else {
+      return NullOpt;
+    }
+  };
+  Stmt new_loop_body =
+      SubstituteAndCollectOpaqueBlock(loop->body, &opaque_block_reuse, substitute_function);
+  for (size_t i = 0; i < inferred_factors.size(); i++) {
+    analyzer.Bind(new_loop_vars[i], Range::FromMinExtent(0, inferred_factors[i]));
+  }
+  // Step 4. Update predicate to guard the loop
+  PrimExpr predicate = substitute_value < loop->extent;
+  new_loop_body = PredicateUpdater(predicate, &analyzer)(new_loop_body);
+  // Step 5. Generate tnested loops to replace the original loop and simplify the binding
+  Stmt outer_stmt = new_loop_body;
+  for (int i = inferred_factors.size() - 1; i >= 0; i--) {
+    outer_stmt = For(new_loop_vars[i], 0, inferred_factors[i], loop->kind, outer_stmt);
+  }
+
+  outer_stmt =
+      Downcast<For>(SimplifyBindings(outer_stmt, GetLoops(loop_sref), &opaque_block_reuse));
+  self->Replace(loop_sref, outer_stmt, opaque_block_reuse);
+  Array<StmtSRef> result_srefs;
+  result_srefs.reserve(inferred_factors.size());
+  for (size_t i = 0; i < inferred_factors.size(); i++) {
+    result_srefs.push_back(self->stmt2ref.at(outer_stmt.get()));
+    const ForNode* outer_loop = outer_stmt.as<ForNode>();
+    ICHECK(outer_loop);
+    outer_stmt = outer_loop->body;
+  }
+  return result_srefs;
+}
+
+StmtSRef Fuse(ScheduleState self, const Array<StmtSRef>& loop_srefs) {
+  //     Invariance
+  //   - The total repeat number has not changed for each direct child block.
+  //   - The execution order has not changed. (The block executes with the same
+  //     args and the same order with before.)
+  std::vector<const ForNode*> loops;
+  loops.reserve(loop_srefs.size());
+  StmtSRef outer_sref{nullptr};
+  const ForNode* outer_loop = nullptr;
+  arith::Analyzer analyzer;
+  // Step 1. check correctness
+  GetScopeRootAndCheckStagePipeline(self, loop_srefs[0]);
+  for (const StmtSRef& sref : loop_srefs) {
+    const auto* loop = sref->StmtAs<ForNode>();
+    if (loop == nullptr) {
+      throw NotLoopError(self->mod, sref->stmt->GetTypeKey());
+    }
+    if (!loop->annotations.empty()) {
+      throw HasAnnotationError(self->mod, GetRef<For>(loop));
+    }
+    if (loop->thread_binding.defined()) {
+      throw HasThreadBindingError(self->mod, GetRef<For>(loop));
+    }
+    if (outer_sref.defined()) {
+      if (sref->parent != outer_sref.get()) {
+        throw OuterNotInnerParent(self->mod, GetRef<For>(outer_loop), GetRef<For>(loop));
+      }
+      Array<Stmt> outer_children = GetChildren(GetRef<Stmt>(outer_loop));
```

Comment:
you probably don't need to use a heavy GetChildren to do the actual checking.

Just check if `outer_loop->body.same_as(loop)` will suffice i suppose